platforms/common/i2c_spi_buses.cpp

```diff
@@ -384,7 +384,7 @@ int BusInstanceIterator::runningInstancesCount() const
 	int num_instances = 0;
 
 	for (const auto &modules : i2c_spi_module_instances) {
-		if (strcmp(modules->_module_name, _module_name) == 0) {
+		if (modules->_bus == bus() && strcmp(modules->_module_name, _module_name) == 0) {
```

Comment:
I think we don't want to break this so that the instance numbering is across all instances on all buses.

How does it break i2c launcher?

Reply:
It shows modules running on other buses that we're not interested in. And `runningInstancesCount()` isn't used anywhere if I checked that right.

Reply:
Here: https://github.com/PX4/PX4-Autopilot/blob/c0609f3df6a2add0c63c7abc8875d01c2b7e2a38/platforms/common/i2c_spi_buses.cpp#L628